### PR TITLE
feat: blacklist e2e tests for fiattokenfactory

### PR DESCRIPTION
**Pull request by boojamya**
Includes:
- `TestFiatTFUpdateBlacklister`
- `TestFiatTFBlacklist`
- `TestFiatTFUnblacklist`

### CODE REVIEW

File: e2e/fiat_tf_test.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble)
+	toBlacklist2 := w[0]
+	alice := w[1]
+
+	res, _, err := val.ExecQuery(ctx, "fiat-tokenfactory", "list-blacklisted")
+	require.NoError(t, err, "failed to query list-blacklisted")
+
+	var preFailedBlacklist, postFailedBlacklist fiattokenfactorytypes.QueryAllBlacklistedResponse
+	_ = json.Unmarshal(res, &preFailedBlacklist)
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go:175]: Add in: [`b634d34` (#370)](https://github.com/noble-assets/noble/pull/370/commits/b634d3409c8133e8a672fbe04e6090a1cdac0706)

File: e2e/fiat_tf_test.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	// EXPECTED: Request fails; user remains unblacklisted
+	// Status:
+	// 	not blacklisted: blacklistedUser1
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Blacklister.KeyName(), "fiat-tokenfactory", "unblacklist", blacklistedUser1.FormattedAddress())
+	require.ErrorContains(t, err, "the specified address is not blacklisted")
+
+	_, err = showBlacklisted(ctx, val, blacklistedUser1)
+	require.Error(t, err, "query succeeded, blacklisted account should not exist")
+}
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go:297]: Added in: [`06a3995` (#370)](https://github.com/noble-assets/noble/pull/370/commits/06a399545b1cef34ac3ceba4ba99e36113eec301)

File: e2e/fiat_tf_test.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+	showBlacklistedRes, err := showBlacklisted(ctx, val, toBlacklist1)
+	require.NoError(t, err, "failed to query show-blacklisted")
+	expectedBlacklistResponse := fiattokenfactorytypes.QueryGetBlacklistedResponse{
+		Blacklisted: fiattokenfactorytypes.Blacklisted{
+			AddressBz: toBlacklist1.Address(),
+		},
+	}
+	require.Equal(t, expectedBlacklistResponse.Blacklisted, showBlacklistedRes.Blacklisted)
+}
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go:216]: Added in: [`06a3995` (#370)](https://github.com/noble-assets/noble/pull/370/commits/06a399545b1cef34ac3ceba4ba99e36113eec301)

File: e2e/fiat_tf_test.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+	showBlacklisterRes, err = showBlacklister(ctx, val)
+	require.NoError(t, err, "failed to query show-blacklister")
+	expectedGetBlacklisterResponse = fiattokenfactorytypes.QueryGetBlacklisterResponse{
+		Blacklister: fiattokenfactorytypes.Blacklister{
+			Address: string(newBlacklister3.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetBlacklisterResponse.Blacklister, showBlacklisterRes.Blacklister)
+}
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go:126]: Added in: [`06a3995` (#370)](https://github.com/noble-assets/noble/pull/370/commits/06a399545b1cef34ac3ceba4ba99e36113eec301)

File: e2e/fiat_tf_test.go [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+	// ACTION: Blacklist user while TF is paused
+	// EXPECTED: Success; user blacklisted
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "to-blacklist-1", math.OneInt(), noble)
+	toBlacklist1 := w[0]
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, toBlacklist1)
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go]: Hi @xngln , we do the check in the blacklistAccount function (https://github.com/noble-assets/noble/blob/149a7f6f8748b761bc2ce2108da09c27916b101a/e2e/utils.go#L392-L402)

File: e2e/fiat_tf_test.go [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+	// Blacklist an account while the blacklister is blacklisted
+	// EXPECTED: Success; user blacklisted
+	// Status:
+	// 	blacklisted: toBlacklist1
+	// 	not blacklisted: toBlacklist2
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Blacklister)
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, toBlacklist2)
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go]: See: https://github.com/noble-assets/noble/pull/370#discussion_r1668197962

File: e2e/fiat_tf_test.go [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+	// EXPECTED: Success; user unblacklisted
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "blacklist-user-1", math.OneInt(), noble)
+	blacklistedUser1 := w[0]
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, blacklistedUser1)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, blacklistedUser1)
```
[the remark this answers is not in the document]
Comment (boojamya) [reply e2e/fiat_tf_test.go:249]: We do this check inside of the unblacklistAccount function: https://github.com/noble-assets/noble/blob/149a7f6f8748b761bc2ce2108da09c27916b101a/e2e/utils.go#L411-L412

**Review by kaliweh-circle — APPROVED** [review]

**Review by johnletey — APPROVED** [review]
LGTM! Passing on Cosmos SDK v0.50 ⚡

- ✅ Green CI for `TestFiatTFUpdateBlacklister` ([link](https://github.com/noble-assets/noble/actions/runs/11381983733/job/31664708087?pr=370))
- ✅ Green CI for `TestFiatTFBlacklist` ([link](https://github.com/noble-assets/noble/actions/runs/11381983733/job/31664708401?pr=370))
- ✅ Green CI for `TestFiatTFUnblacklist` ([link](https://github.com/noble-assets/noble/actions/runs/11381983733/job/31664708718?pr=370))

**Pull request merged**